src/docbuild/models/config_model/app.py

```diff
@@ -1,79 +1,140 @@
 """Pydantic model for application configuration."""
 
-from typing import Any, Self
-from pydantic import BaseModel, Field, model_validator, ConfigDict
 from copy import deepcopy
+from typing import Any, Self, Literal, Optional, TYPE_CHECKING
+import logging # Needed for DEFAULT_LOGGING_CONFIG usage
+
+from pydantic import BaseModel, Field, model_validator, ConfigDict, root_validator # Added root_validator for clarity
 
 from docbuild.config.app import replace_placeholders
 from docbuild.config.app import CircularReferenceError, PlaceholderResolutionError
-from docbuild.logging import DEFAULT_LOGGING_CONFIG
+from docbuild.logging import DEFAULT_LOGGING_CONFIG # Note: Must import the actual DEFAULT_LOGGING_CONFIG
 
-# 1. Logging Sub-Model
-class App_LoggingConfig(BaseModel):
-    """Configuration for the application's logging system.
+# Ensure type annotations work correctly even if type aliases are not natively available
+if TYPE_CHECKING:
+    from typing import TypeAlias
+else:
+    TypeAlias = Any
+
+
+# --- 1. Logging Sub-Models (Schema for logging.dictConfig) ---
+
+class FormatterConfig(BaseModel):
+    """
+    Defines the configuration for a single logging formatter.
+    Allows extra keys for custom class/factory arguments.
+    """
+    # Allows for custom factory parameters and the special '()' key.
+    model_config = ConfigDict(extra='allow', populate_by_name=True) 
+
+    format: str | None = None
+    datefmt: str | None = None
+    style: Literal['%', '{', '$'] | None = None
+    class_name: str | None = Field(None, alias='class') 
```

Comment:
Could you add a `description` key here?


```suggestion
    class_name: str | None = Field(None, alias='class',
                                   description='The fully qualified name of the handler class',
                                   ) 
```